components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityUtil.java

```diff
@@ -355,36 +357,58 @@ public static String getEndpointURIPath(String endpoint, boolean addProxyContext
         return serverUrl.toString();
     }
 
-    public static String getServerURL(String endpoint, boolean addProxyContextPath, boolean addWebContextRoot)
+    /**
+     * This util method is used to construct a complete URL out of the given URL context.
+     * @param urlContext URL context.
+     * @param addProxyContextPath add proxy context path to the URL.
+     * @param addWebContextRoot add web context path to the URL.
+     * @return complete URL for the given URL context.
+     * @throws IdentityRuntimeException if error occurred while constructing the URL
+     */
+    public static String getServerURL(String urlContext, boolean addProxyContextPath, boolean addWebContextRoot)
             throws IdentityRuntimeException {
-        String hostName = ServerConfiguration.getInstance().getFirstProperty(IdentityCoreConstants.HOST_NAME);
 
+        return getServerURL(urlContext, addProxyContextPath, addWebContextRoot, false);
+    }
+
+    /**
+     * This util method is used to resolve a URL.
```

Comment:
What do we mean by "resolving" a URL?
I think explaining that would indicate the purpose of this method.

Reply:
Resolving means returning a URL with a proxy context path, a web context root and the tenant domain (If required) when provided with a URL context. Will add this.